docs/src/modules/components/Carbon.js

```diff
@@ -80,16 +102,82 @@ class Carbon extends React.Component {
     if (ad) {
       ad.appendChild(script);
     }
+
+    this.checkAdblock();
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timerAdblock);
   }
 
+  timerAdblock = null;
+
+  checkAdblock = (attempt = 1) => {
+    if (document.querySelector('#carbonads')) {
+      this.setState({
+        adblock: false,
+      });
+      return;
+    }
+
+    if (attempt < 3) {
+      this.timerAdblock = setTimeout(() => {
+        this.checkAdblock(attempt + 1);
+      }, 500);
+    } else {
+      this.setState({
+        adblock: true,
+      });
+    }
+  };
+
   render() {
     const { classes } = this.props;
-    return <div className={classes.ad} id="ad" />;
+    const { adblock } = this.state;
+
+    if (adblock) {
+      return (
+        <div className={classes.root}>
+          <div className={classes.ad}>
+            <div id="carbonads">
+              <Typography gutterBottom>Like Material-UI?</Typography>
+              <Typography gutterBottom>
+                {`If you don't mind tech-related ads, and want to support Open Source, whitelist
+                Material-UI in your ad blocker.`}
```

Comment:
Is there room in the available ad-space for the word "please"?

"If you don't mind tech-related ads, and want to support Open Source, whitelist
please Material-UI in your ad blocker.

Reply:
The text is already quite long :/.

![capture d ecran 2018-01-01 a 23 29 31](https://user-images.githubusercontent.com/3165635/34471498-a4c61088-ef4b-11e7-87aa-05231b439eaa.png)

Reply:
Maybe, but it could make a positive difference. (Now you'll have to A/B test! 😆 )

Reply:
Ok, let's try then :).